tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,273 @@
+use sharded_slab::{Guard, Slab};
+
+use super::stack::SpanStack;
+use crate::{
+    registry::{
+        extensions::{Extensions, ExtensionsInner, ExtensionsMut},
+        LookupSpan, SpanData,
+    },
+    sync::RwLock,
+};
+use std::{
+    cell::RefCell,
+    sync::atomic::{fence, AtomicUsize, Ordering},
+};
+use tracing_core::{
+    dispatcher,
+    span::{self, Current, Id},
+    Event, Interest, Metadata, Subscriber,
+};
+
+/// A shared, reusable store for spans.
+///
+/// A `Registry` is a [`Subscriber`] around which multiple [`Layer`]s
+/// implementing various behaviors may be [added]. Unlike other types
+/// implementing `Subscriber` `Registry` does not actually record traces itself:
+/// instead, it collects and stores span data that is exposed to any `Layer`s
+/// wrapping it through implementations of the [`LookupSpan`] and
+/// [`LookupMetadata`] traits. The `Registry` is responsible for storing span
+/// metadata, recording relationships between spans, and tracking which spans
+/// are active and whicb are closed. In addition, it provides a mechanism
+/// `Layer`s to store user-defined per-span data, called [extensions], in the
+/// registry. This allows `Layer`-specific data to benefit from the `Registry`'s
+/// high-performance concurrent storage.
+///
+/// This registry is implemented using a [lock-free sharded slab][slab], and is
+/// highly optimized for concurrent access.
+///
+/// [slab]: https://docs.rs/crate/sharded-slab/
+/// [`Subscriber`]:
+///     https://docs.rs/crate/tracing-core/latest/tracing_core/subscriber/trait.Subscriber.html
+/// [`Layer`]: ../trait.Layer.html
+/// [added]: ../trait.Layer.html#method.with_subscriber
+/// [`LookupSpan`]: trait.LookupSpan.html
+/// [`LookupMetadata`]: trait.LookupMetadata.html
+/// [extensions]: extensions/index.html
+#[derive(Debug)]
+pub struct Registry {
+    spans: Slab<DataInner>,
+}
+
+/// Span data stored in a [`Registry`].
+///
+/// The registry stores well-known data defined by tracing:
+/// span relationships, metadata and reference counts.
+/// Additional user-defined data provided by [Layers],
+/// such as formatted fields, metrics, or distributed traces
+/// should be stored in the [extensions] typemap.
```

Comment:
why is this wrapped at column 60 rather than column 80? are you doing this just to piss me off?

Reply:
_should_ be fixed with that extension you showed me.